src/passes/AlignmentLowering.cpp

```diff
@@ -125,21 +122,16 @@ struct AlignmentLowering : public WalkerPass<PostWalker<AlignmentLowering>> {
     } else {
       WASM_UNREACHABLE("invalid size");
     }
-    replaceCurrent(
-      builder.makeBlock({builder.makeLocalSet(temp, curr->ptr), ret}));
+    return builder.makeBlock({builder.makeLocalSet(temp, curr->ptr), ret});
   }
 
-  void visitStore(Store* curr) {
+  // Core lowering of a 32-bit store. Other storess are done using this.
+  Expression* lowerStoreI32(Store* curr) {
     if (curr->align == 0 || curr->align == curr->bytes) {
-      return;
+      return curr;
     }
```

Comment:
It looks like this could be removed or turned into an assertion, since it is also checked below in `visitStore` before this function is called. The same goes for the analogous check in `lowerLoadI32`.

Reply:
Hmm, `visitStore` does check this initially, but then it calls `lowerStoreI32` from multiple places, and some of the subcalls might be aligned (e.g. a 64-bit load of alignment 4 would have sub-loads that are aligned). So it can't be removed here I don't think.

We might remove the check from `visitStore`, which would mean we always call `lowerStoreI32` even for aligned stores. That's less code, but also less clear maybe - WDYT?

Reply:
Oh I see. Probably fine to keep it then.

Reply:
Thinking some more, it's better to be less surprising, I'll remove the extra checks.